tests/FunctionalTestCase.php

```diff
@@ -252,33 +260,65 @@ public static function getModuleInfo(string $row): ?string
     }
 
     /**
-     * Creates the test collection with the specified options.
+     * Creates the test collection with the specified options  and ensures it is dropped again during tearDown().
      *
      * If the "writeConcern" option is not specified but is supported by the
      * server, a majority write concern will be used. This is helpful for tests
      * using transactions or secondary reads.
      */
-    protected function createCollection(array $options = []): void
+    protected function createCollection(?string $databaseName = null, ?string $collectionName = null, array $options = []): void
     {
         $options += ['writeConcern' => new WriteConcern(WriteConcern::MAJORITY)];
 
-        $operation = new CreateCollection($this->getDatabaseName(), $this->getCollectionName(), $options);
+        if (null === $databaseName) {
+            $databaseName = $this->getDatabaseName();
+        }
+
+        if (null === $collectionName) {
+            $collectionName = $this->getCollectionName();
+        }
+
+        $operation = new CreateCollection($databaseName, $collectionName, $options);
         $operation->execute($this->getPrimaryServer());
+        $this->collectionsToCleanup[$databaseName][$collectionName] = true;
     }
 
     /**
-     * Drops the test collection with the specified options.
-     *
-     * If the "writeConcern" option is not specified but is supported by the
-     * server, a majority write concern will be used. This is helpful for tests
-     * using transactions or secondary reads.
+     * Drops the test collection and ensures it is dropped again during tearDown().
      */
-    protected function dropCollection(array $options = []): void
+    protected function dropCollection(?string $databaseName = null, ?string $collectionName = null): void
```

Comment:
I am tempted to reverse `$databaseName` & `$collectionName`. Both are optional, but `$collectionName` is more often modified than `$databaseName`.

Reply:
This seems fine, as I can't think of any places in the library where both a database and collection are accepted and we use that reverse order. Seeing an explicit `null` in the calls will also serve as a reminder that we're using the default database.